tests/test_calc_error.py

```diff
@@ -0,0 +1,382 @@
+                i = 2
+            elif (t < 900):
+                i = 1
+            elif (t < 1800):
+                i = 4
+            elif (t < 3000):
+                i = 2
+            else:
+                i = 3
+            return m.InputContainer({'i': i})
+    
+        simulated_results = m.simulate_to(2000, future_loading, **options)
+
+        # Initializing parameters to very erroneous values       
+        m.parameters['qMax'] = 4000
+
+        # With our current set parameters, our model goes unstable immediately
+        with self.assertRaises(ValueError) as cm:
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=1)
+        self.assertEqual(
+            "Model unstable- NAN reached in simulation (t=0.0) before cutoff threshold. Cutoff threshold is 1900.0, or roughly 95.0% of the data",
+            str(cm.exception)
+        ) 
+
+        # Creating duplicate model to check if consistent results occur
+        m1 = BatteryElectroChemEOD()
+
+        # Much bigger parameter initialization
+        m.parameters['kp'] = m1.parameters['kp'] = 10000
+        m.parameters['kn'] = m1.parameters['kn'] = 1000 
+        m.parameters['qpMax'] = m1.parameters['qpMax'] = 4500
+        m.parameters['qMax'] = m1.parameters['qMax'] = 9000
+
+        simulated_results = m.simulate_to(2000, future_loading, **options)
+        m1_sim_results = m1.simulate_to(2000, future_loading, **options)
+        
+        # Checks to see if model goes unstable before default stability tolerance is met.
+        with self.assertRaises(ValueError) as cm:
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt = 1)
+        self.assertEqual(
+            "Model unstable- NAN reached in simulation (t=1800.0) before cutoff threshold. Cutoff threshold is 1900.0, or roughly 95.0% of the data",
+            str(cm.exception)
+        )
+        
+        # Checks to see if m1 throws the same exception. 
+        with self.assertRaises(ValueError):
+            m1.calc_error(m1_sim_results.times, m1_sim_results.inputs, m1_sim_results.outputs, dt = 1)
+        self.assertEqual(
+            "Model unstable- NAN reached in simulation (t=1800.0) before cutoff threshold. Cutoff threshold is 1900.0, or roughly 95.0% of the data",
+            str(cm.exception)
+        )
+
+        # Checks to see if stability_tolerance throws Warning rather than an Error when the model goes unstable after threshold
+        with self.assertWarns(UserWarning) as cm:
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, 
+                     dt = 1, stability_tol=0.7)
+        self.assertEqual(
+            'Model unstable- NaN reached in simulation (t=1800.0)',
+            str(cm.warning)
+        )
+
+    def test_MSE(self):
+        """
+        Base tests that ensure first-level workability within calc_error
```

Comment:
I'm not sure I understand this comment, but it might be software engineering specific language that I'm unfamiliar with